### PR TITLE
Qol 9451 fix startup issues on small instances

### PR DESCRIPTION
**Pull request by duttonw**
fix swap,

### CODE REVIEW

File: recipes/swapfile.rb [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
-# Author:: Carl Antuar (<carl.antuar@smartservice.qld.gov.au>)
+# Author:: Carl Antuar (<carl.antuar@smartservice.qld.gov.au>), William Dutton (<will.dutt@gmail.com>)
 # Cookbook Name:: datashades
 # Recipe:: swapfile
 #
-# Sets up a swapfile on the extra EBS, if available.
-# Since this isn't on the root filesystem and may not be instantly
-# available on startup, it's not added to /etc/fstab.
+# Sets up a swapfile on the root disk
+# Setup another swap on external disk for more speed if it is used during normal operations
```
Comment (ThrawnCA) [inline recipes/swapfile.rb:7]: It's not an extra disk anymore, it's the root.

File: recipes/swapfile.rb [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+# Since extra disk is attached on stage one chef, it is dynamically added on opsworks setup
 #
-# Copyright 2021, Queensland Government
+# Ensure root disk is bigger than 8gb else issues will happen down the line, i.e. 100gb per below:
```
Comment (ThrawnCA) [inline recipes/swapfile.rb:10]: Should be eg not ie.

File: recipes/swapfile.rb [hunk 3]
```diff
@@ -22,12 +36,35 @@
...
+  fstype 'swap'
+end
+
+bash 'activate all swap devices' do
+  user 'root'
+  code 'swapon -a'
+end
+
 extra_disk = "/mnt/local_data"
 if ::File.directory?(extra_disk) then
```
Comment (ThrawnCA) [inline recipes/swapfile.rb:62]: This should go away.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

File: recipes/swapfile.rb (hunk 1 above)
Comment (duttonw) [reply recipes/swapfile.rb:7]: I'm still adding the extra swap for extra performance

https://wiki.archlinux.org/title/swap

If two or more areas have the same priority, and it is the highest priority available, pages are allocated on a round-robin basis between them.

Comment (ThrawnCA) [reply recipes/swapfile.rb:7]: I don't see how that will improve performance? A 4GB swapfile is already overkill for every use case we've encountered. Adding even more than that seems pointless.

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**